### PR TITLE
SHUSH Attack

### PR DESCRIPTION
**Pull request by vijay-jaisankar**
## Why ?
This PR introduces the SHUSH attack, a simple and fast  algorithm that modifies the input audio tensor by setting a fraction of its indices to 0. Researchers can test out Audioseal models under different settings of the SHUSH attack for their datasets and use-cases.

## How ?
The SHUSH attack takes in a tensor and sets `K` of its first indices, i.e `tensor[:K]` to `0.0`, and is a synthetic alternative to audio corruption or third-party masking. This implementation allows `K` to be set through parameters, hence researchers can make the attack more severe or less severe for their analysis.

## Test plan
We tried different variants of the attack on the [RAVDESS Emotional Speech Audio](https://www.kaggle.com/datasets/uwrfkaggler/ravdess-emotional-speech-audio) dataset. The corresponding Kaggle notebook can be found [here](https://www.kaggle.com/code/vijayjaisankar/audioseal-shush-attack-ravdess-benchmarking). Audioseal shows strong performance in both realistic settings of low tampering and settings with high tampering attacks.

### CODE REVIEW

File: examples/attacks.py [hunk 1]
```diff
@@ -265,3 +265,29 @@ def identity(
...
+        Returns:
+        - torch.Tensor: Transformed audio tensor.
+        """
+        batch_size, channels, time = tensor.shape
+        shush_tensor = tensor.detach().clone()
+        
+        # Set the first `fraction*time` indices of the waveform to 0
+        for b in range(batch_size):
+            for channel in range(channels):
+                shush_tensor[b][channel][0:int(fraction*time)] = 0.0
```
Comment (antoine-tran) [inline examples/attacks.py:291]:
```suggestion
        time = tensor.size(-1)
        shush_tensor = tensor.detach().clone()
        
        # Set the first `fraction*time` indices of the waveform to 0
        shush_tensor[:, :, :int(fraction*time)] = 0.0
```

**Review by antoine-tran — COMMENTED** [review]
Hi @vijay-jaisankar , thanks a lot for your contribution !

Your PR looks good to me. I made a small suggestion on minor optimization.

Also, it would be nice if you can update our notebook, or create a new one with the example you have in Kaggle on RAVDESS datasets. That would be a great resource for the community

**Comment by vijay-jaisankar** [comment]
Thanks a lot for the suggestions!

Will push the optimisation in the review with a consolidated version of the notebook for review soon in the coming days.

File: examples/attacks.py (hunk 1 above)
Comment (vijay-jaisankar) [reply examples/attacks.py]: This has been added in https://github.com/facebookresearch/audioseal/pull/25/commits/f39691178b36963665a50c2a7d1f98acf9dddc7e

**Comment by antoine-tran** [comment]
@vijay-jaisankar Apologies if I sound nitpicking here, but could you remove the copy-paste code of attacks.py in your note book and import it instead ? I guess that was the whole point of migrating your code to audioseal :)

**Comment by vijay-jaisankar** [comment]
Hello @antoine-tran , sorry, I missed making that change in the previous commit. The code has been replaced with the import statement accordingly in https://github.com/facebookresearch/audioseal/pull/25/commits/6a5bcea4129847d27201d7c2dea7bb358bd039ce, please check.

Thanks!

**Review by antoine-tran — APPROVED** [review]
Thanks a lot for your contribution !
